pyro/contrib/funsor/infer/trace_elbo.py

```diff
@@ -21,29 +22,68 @@ def differentiable_loss(self, model, guide, *args, **kwargs):
         with enum(), plate(
             size=self.num_particles
         ) if self.num_particles > 1 else contextlib.ExitStack():
-            guide_tr = trace(config_enumerate(default="flat")(guide)).get_trace(
-                *args, **kwargs
-            )
+            guide_tr = trace(
+                config_enumerate(default="flat", num_samples=self.num_particles)(guide)
```

Comment:
Without `num_samples` argument here `config_enumerate` wasn't adding `{"enumerate": "flat"}` to `msg["infer"]`.